pkg/options/types.go

```diff
@@ -141,15 +140,17 @@ func (n *NodeType) Type() string {
 
 // GetNodeFieldSelector returns a nodename field selector.
 func (n *NodeType) GetNodeFieldSelector() string {
-	if nil == n || len(*n) == 0 {
+	if nil == n {
```

Comment:
To make the test pass and keep most of the logic, I think we can just revert this line.

```suggestion
	if nil == n || len(*n) == 0 {
```

I was trying to look at the usage of this function in [pkg/app/server.go#L208](https://github.com/kubernetes/kube-state-metrics/blob/7995d5fd23bcff7ae24ab6849f7c393d262fb025/pkg/app/server.go#L208).

... However, I still don't fully understand what the intended behavior is when `--node` is not provided.

Reply:
After reading https://github.com/kubernetes/kube-state-metrics/pull/2359 I believe this is the correct fix.